sw/device/lib/base/macros.h

```diff
@@ -441,6 +441,31 @@
 /**
  *  This macro is used to align an offset to point to a 32b value.
  */
-#define OT_ALIGN_MEM(x) (uint32_t)(4 + (((uintptr_t)(x)-1) & ~3))
+#define OT_ALIGN_MEM(x) (uint32_t)(4 + (((uintptr_t)(x)-1) & ~3u))
 
+struct OtSignConversionUnsupportedType {
+  char err;
+};
+
+/**
+ * This macro converts a given unsigned integer value to its signed counterpart.
+ */
+#define OT_SIGNED(value)          \
+  _Generic((value),               \
+      uint8_t: (int8_t)(value),   \
+      uint16_t: (int16_t)(value), \
+      uint32_t: (int32_t)(value), \
+      uint64_t: (int64_t)(value), \
+      default: (struct OtSignConversionUnsupportedType){.err = 1})
```

Comment:
> @luismarques I don't believe it's possible to use `static_assert` here because it will be evaluated regardless of which type matches `value`. Dynamic `CHECK` is undesirable because this is a compile-time error. So I introduced a struct (must have one member due to `-Wgnu-empty-struct`) that we can use when we want to emit a value that cannot be interpreted as an integer. WDYT?

Sounds good! That's the sort of thing I meant with the "some kind of" in `some_kind_of_assert` :)